src/transit_equity/networks/network_prep.py

```diff
@@ -0,0 +1,361 @@
+
+    load_wkb(x):
+        Decodes and loads WKB binary location data into a Shapely geometry object for plotting with 
+        GeoPandas.
+        Args:
+            x (object): WKB binary object.
+        Returns:
+            object: Shapely geometry object.
+
+Example usage:
+    import pandas as pd
+    import geopandas as gpd
+    from transit_equity.networks.network_prep import clean_and_filter_network_data, 
+                                                        get_hex_centroids_for_od, 
+                                                        load_wkb
+
+    # Load trip data
+    trips_df = pd.read_sql(query.statement,         #sql alchemy query 
+                            engine_ng)              #sql alchemy database engine
+
+    # Clean and filter trip data
+    gdf_trips = clean_and_filter_network_data(trips_df)
+
+    # Define path to hex grid shapefile
+    hex_grid_path = 'path_to_hex_grid.shp'
+
+    # Map origin-destination pairs to hexagon centroids and calculate frequencies
+    gdf_network_clean = get_hex_centroids_for_od(gdf_trips, hex_grid_path)
+"""
+
+import binascii
+import pandas as pd
+import geopandas as gpd
+from shapely import wkb
+
+def load_wkb(x):
+    """Function to decode and load WKB binary location
+    into Shapely geometry object to enable plotting
+    with geoPandas.
+    
+    Parameters
+    ----------
+    hex_string : object
+        WKB binary object
+
+    Returns
+    -------
+    object
+        Shapely geometry object.
+
+    """
+    return wkb.loads(binascii.unhexlify(x))
+
+def clean_and_filter_network_data(trips_df):
+    """
+    Cleans and filters trip data, transforming it into a GeoDataFrame for spatial analysis.
+
+    Args:
+    trips_df (pd.DataFrame): DataFrame containing trip data with columns:
+        - 'stop_location': Boarding location as a binary string.
+        - 'stop_location_1': Alighting location as a binary string.
+        - 'device_dtm_pacific': Boarding date and time.
+        - 'alight_dtm_pacific': Alighting date and time.
+        - 'txn_id': Boarding transaction ID.
+        - 'txn_id_1': Alighting transaction ID.
+        - 'card_id': ID of the card used for the trip.
+
+    Returns:
+    gpd.GeoDataFrame: Cleaned GeoDataFrame with columns:
+        - 'card_id': ID of the card used for the trip.
+        - 'board_location': Original boarding location binary string.
+        - 'alight_location': Original alighting location binary string.
+        - 'board_location_shapely': Shapely geometry of the boarding location.
+        - 'alight_location_shapely': Shapely geometry of the alighting location.
+        - 'trip_time_minutes': Duration of the trip in minutes.
+        - 'trip_frequency': Frequency of trips between the boarding and alighting locations.
+        - 'board_string': String representation of the boarding location.
+        - 'alight_string': String representation of the alighting location.
+
+    Steps:
+        1. Rename columns for clarity.
```

Comment:
I like the steps comment! 
We can use it later to make this method more flexible and modular, such as converting to a class and making a method out of each step.